lang/cs/Org.Apache.REEF.Network/Group/Driver/Impl/GroupCommunicationMessage.cs

```diff
@@ -17,91 +17,191 @@
  * under the License.
  */
 
+using System;
+using System.Threading;
+using Org.Apache.REEF.Network.StreamingCodec;
+using Org.Apache.REEF.Wake.Remote;
+using Org.Apache.REEF.Tang.Annotations;
+
 namespace Org.Apache.REEF.Network.Group.Driver.Impl
 {
     /// <summary>
-    /// Messages sent by Group Communication Operators
+    /// Messages sent by MPI Operators. This is the Writable version of GroupCommunicationMessage
+    ///  class and will eventually replace it once everybody agrees with the design
     /// </summary>
-    public class GroupCommunicationMessage
+    [Obsolete("Need to remove Iwritable and use IstreamingCodec. Please see Jira REEF-295 ", false)]
+    public sealed class GroupCommunicationMessage<T> : GeneralGroupCommunicationMessage
     {
+        private readonly IStreamingCodec<T> _codec;
+
+        /// <summary>
+        /// Empty constructor to allow instantiation by reflection
+        /// </summary>
+        [Inject]
+        private GroupCommunicationMessage(IStreamingCodec<T> codec)
```

Comment:
Can yo u crate a message instance without other information (like those in other constructors)?